### PR TITLE
Release/banking bolt

### PR DESCRIPTION
**Pull request by MarcoBgn**
@cesar-tonnoir , as discussed

### CODE REVIEW

**Comment by cesar-tonnoir** [comment]
@MarcoBgn usually we don't commit the dist files: the project should be built when we prepare a release. I'll delete your last commit before merging

**Comment by cesar-tonnoir** [comment]
Ok, so actually I cannot just delete the last commit because you've deleted the dist files in your first commit (#3930dce).
I'll merge your pull request to a temporary branch.
Also why do you have a new "average purchase size" widget there? I don't think this is related to the _Kappa_ delivery?

File: src/components/widgets-settings/bolt-account/bolt-account.directive.coffee [hunk 1]
```diff
@@ -0,0 +1,66 @@
+module = angular.module('impac.components.widgets-settings.bolt-account',[])
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/bolt-account/bolt-account.directive.coffee]: Can you review the naming please? "Bolt account" for a widget setting is not self-explanatory. Can you please reming me why we need a new setting here (versus reusing/improving the existing "account" setting?)

File: src/components/widgets-settings/bolt-account/bolt-account.less [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.analytics .widget-item .content .settings.select-account {
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/bolt-account/bolt-account.less]: name of the css class if different from the name of the setting?

File: src/components/widgets-settings/bolt-account/bolt-account.tmpl.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<h5 ng-if="showLabel">{{label}}</h5>
+<div class="input-group settings select-account">
+	<select ng-model="parentWidget.selectedAccount" ng-change="onAccountSelected()" ng-options="formatLabel(account) for account in parentWidget.accountList track by account.name" class="form-control">
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/bolt-account/bolt-account.tmpl.html]: You shouldn't use `parentWidget` if this can be avoided: pass the `accountList` to your directive as an argument instead, this will keep the concerns separated

File: src/components/widgets-settings/hist-mode/hist-mode.tmpl.html [hunk 4]
```diff
@@ -1,12 +1,12 @@
 <div class="settings hist-mode-choser">
   <div common-time-period-info context="forwardParams" />
 
-  <div align="center" class="options">
+  <div align="center" class="options" ng-show="!parentWidget.isLiveBalance">
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/hist-mode/hist-mode.tmpl.html]: Wow, this is very intrusive: you should pass a parameter to the directive instead, something like "show-mode-choser" which would be true by default

File: src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  ]
+
+  # Widget specific methods
+  # --------------------------------------
+
+  $scope.isDataFound=true
+  w.initContext = ->
+    $scope.isDataFound = w.content?
+
+  $scope.getName = ->
```
Comment (cesar-tonnoir) [inline src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee]: do you need a method there? Can't you display `w.selectedAccount.name` directly in the view (I don't think this would raise an error even if none account is selected)

File: src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    w.selectedAccount.name if w.selectedAccount?
+
+  $scope.getTitle = ->
```
Comment (cesar-tonnoir) [inline src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee]: same comment here and for the balances below

File: src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    ) if w.content?
+    _.find(statement.table_rows, (row) ->
+      row.column_2 == "Closing Balance"
+    ) if statement
+
+  $scope.displayAccount = ->
+    $scope.updateSettings(false).then ->
+      w.format()
+
+  # Needed for Kpis compatibility?
```
Comment (cesar-tonnoir) [inline src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee]: Not sure... Do we have any kpi coded in the backend for this widget? I think you can remove this part?..

File: src/components/widgets/accounts-live-balance/accounts-live-balance.tmpl.html [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  Component generated by Impac! Widget Generator!
+-->
+<div widget-accounts-live-balance class="tall-widget">
+  <!-- Settings Panel -->
+  <div ng-show="widget.isEditMode" class="edit">
+    <h4>Widget settings</h4>
+
+    <div setting-organizations parent-widget="widget" class="part" deferred="::orgDeferred" />
+    <div setting-bolt-account parent-widget="widget" on-account-selected="updateKpiExtraParams()" class="part" deferred="::accountBackDeferred" />
+    <!-- <div setting-time-period parent-widget="widget" class="part" deferred="::timePeriodDeferred" hist-params="widget.metadata.hist_parameters" hide-chart-interval="true"/> -->
```
Comment (cesar-tonnoir) [inline src/components/widgets/accounts-live-balance/accounts-live-balance.tmpl.html]: Can the comments be removed?

File: src/components/widgets/sales-average-purchase-size/sales-average-purchase-size.directive.coffee [hunk 9]
```diff
@@ -0,0 +1,112 @@
+#
+# Component generated by Impac! Widget Generator!
```
Comment (cesar-tonnoir) [inline src/components/widgets/sales-average-purchase-size/sales-average-purchase-size.directive.coffee]: This widget seems to be out of scope for this PR

File: src/locales/en-AU.json [hunk 10]
```diff
@@ -69,7 +69,9 @@
   "impac.common.expand":                   "expand",
 
   "impac.data_not_found_config.title":            "Data not found",
+  "impac.data_not_found_config.liveBTitle":       "Xero not found",
```
Comment (cesar-tonnoir) [inline src/locales/en-AU.json]: Snake-case?

File: src/services/highcharts-factory/highcharts-factory.svc.coffee [hunk 11]
```diff
@@ -52,14 +52,14 @@ angular
       @_template = templates[@options.chartType]
       return
 
-    render: (data, options)->
+    render: (data, options, renderer_callback)->
```
Comment (cesar-tonnoir) [inline src/services/highcharts-factory/highcharts-factory.svc.coffee]: I think this is out of scope too - can it be removed?

**Comment by cesar-tonnoir** [comment]
After the changes, could you please open a PR against th ebranch #release/v1.6.9 ?
Once merge, your commits should appear in => https://github.com/maestrano/impac-angular/pull/469

File: src/locales/en-AU.json (hunk 10 above)
Comment (MarcoBgn) [reply src/locales/en-AU.json]: Well spotted, thanks

File: src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee (hunk 5 above)
Comment (MarcoBgn) [reply src/components/widgets/accounts-live-balance/accounts-live-balance.directive.coffee]: Yep, 👍

**Comment by MarcoBgn** [comment]
@cesar-tonnoir opened a clean PR with the changes as discussed #471
